src/cascadia/TerminalSettingsEditor/Utils.h

```diff
@@ -138,3 +138,41 @@ namespace winrt::Microsoft::Terminal::Settings
     winrt::hstring GetSelectedItemTag(winrt::Windows::Foundation::IInspectable const& comboBoxAsInspectable);
     winrt::hstring LocalizedNameForEnumName(const std::wstring_view sectionAndType, const std::wstring_view enumValue, const std::wstring_view propertyType);
 }
+
+// BODGY!
+//
+// The following function and struct are a workaround for GH#9320.
+//
+// DismissAllPopups can be used to dismiss all popups for a particular UI
+// element. However, we've got a bunch of pages with scroll viewers that may or
+// may not have popups in them. Rather than define the same exact body for all
+// their ViewChanging events, the HasScrollViewer struct will just do it for
+// you!
+inline void DismissAllPopups(winrt::Windows::UI::Xaml::XamlRoot const& xamlRoot)

```

Comment:
It's fine to put this in the template class body; it will get emitted for every consumer, yes, but then it will get folded down into a single implementation by COMDAT folding/OPT:ICF. Otherwise, you might need to change the specifiers on this a little bit. Something about inline.

Then again, [someone on the LLVM mailing list](https://lists.llvm.org/pipermail/llvm-dev/2021-August/152031.html) had this to say on the matter:

> C++ `inline` means 'the definition is provided in line with the 
> declaration, the compiler and linker are responsible for ensuring that 
> exactly one definition exists in the final binary'
> ...
> ...
>   - C++ `inline`: Must exist in at least one compilation unit and the 
linker must discard duplicates.

Sounds like what we want. So you can ignore that part. :P